src/internal.c

```diff
@@ -11763,7 +11767,7 @@ int SendUserAuthFailure(WOLFSSH* ssh, byte partialSuccess)
         ret = WS_BAD_ARGUMENT;
 
     if (ret == WS_SUCCESS) {
-        authSz = GetAllowedAuth(ssh, authStr);
+        authSz = GetAllowedAuth(ssh, authStr, &partialSuccess);
```

Comment:
I think the partial success should be coming from the userAuthCb return value. Adding WOLFSSH_USERAUTH_PARTIAL_SUCCESS to the list of possible return values from the user auth instead of to the list of auth types. And then if it is returned, setting a new variable partialSuccess to 1 and passing it as the last argument to SendUserAuthFailure in the DoAuthRequest* functions instead of the current hard set '0' in wolfSSH master.

Reply:
@JacobBarthelmeh the advantage of handling it the current way is that the change is centralized in a few lines of code. 

We can do it the way you suggest, but then it would conflict with PR #559. I suggest that to be merged first, then.